### DISCUSSION TITLE
Can you obtain the server's IP/domain a player pings?

### DISCUSSION THREAD

**Discussion by Andre601** [dmulloy2/ProtocolLib#1784] [Q&A] [answered]
Would like to get whatever adress a player pinged.
If the player f.e. added play.example.com as the domain in their server list and ping it would I like to get that address.

Is this doable with ProtocolLib's `onPacketSending(PacketEvent)`?

**Comment by derklaro** [comment, marked as answer]
It is not possible with `onPacketSending(PacketEvent)`, you need to store the address the player pinged which comes from the handshake (`PacketType.Handshake.Client.SET_PROTOCOL`) as a string.

**Reply by Andre601** [reply]
So would I use the onPacketReceiving for this?
Or do I simply get the PacketType.Handshake.Client.SET_PROTOCOL in the other?

**Reply by Andre601** [reply]
To add a bit more info: I'm using `new PacketAdapter(JavaPlugin, ListenerPriority, PacketType.Status.Server.SERVER_INFO)` right now with the onPacketSending method to modify the server list entry on the client and now want to get the server's address the player used to ping.

**Reply by derklaro** [reply]
Yes, you need to listen to `onPacketReceiving` and store the value, then retreive it again when the status response is sent.

**Reply by Andre601** [reply]
So it is
```java
protocolManager.addPacketListener(new PacketAdabter(plugin, ListenerPriority.LOW, PacketType.Handshake.Client.SET_PROTOCOL){
    @Override
    public void onPacketReceiving(PacketEvent event){
        // Do something here
    }
});
```
...right?

Tho, the question for me remains on how to obtain the pinged server address here. I'm not too familiar with the PacketEvent so idk the structure it would have here and what I need to do to actually get the address.

**Reply by derklaro** [reply]
Basically you retrive packet data by going by their field type. So you would need to use `event.getPacket().getStrings().read(/* index */ 0)` to get the first String field.

**Reply by Andre601** [reply]
I see. Thanks. Will try it out and let you know about the outcome

**Reply by Andre601** [reply]
It works like a charm. Thanks fot the help!
